### PR TITLE
Refactor the combined search implementation with form and search adapters

### PR DESCRIPTION
**Pull request by nikomakela**
LIIKUNTA-425.

### Combined search

The search page has 1 form, but that form triggers searches to multiple datasources:

1. The venue search tab makes a query to the Unified-Search.
2. The events and the course searches makes a query to the LinkedEvents.

This means that:

1. Parameters given as an input to the form should adapt to search from different datasources. 1 input can be called “q“ in another datasource, but “text“ in another.
2. Not all the parameters are used in every search, because not all the datasources have an equivalent filter.

#### Input: The URL (search) parameters

_The URL is the primary source for input and state._

The URL controls the whole search form, for easy maintain and also for the accessibility reasons:

1. The URL should be updated when new filters are applied (or any of them are removed / cleared). The active filters are always shown in the UI.
2. When another search tab is clicked, the search URL should not change from the search params.
3. When a filter is activated, the URL should be updated. When an URL is updated, the form should also ract to that.

An example of CombinedSearchAdapterInput type, that could be used as an input for the searches:

```typescript
// NOTE: this is just an example, not an actual implementation
type CombinedSearchAdapterInput = {
  text: string;
  orderBy: string;
  sportsCategories: string[];
  organization: string | null;
  keywords: string[];
};
```

The input URL should also be cleaned for the combined search. A conversion map can handle any needed _backward support_, but also to full fill any possible _mandatory inputs_, that should always exists there. It can also _initialize the search form_. It could convert some params to some others and initialize the form with the transformed keys and values.

E.g. `/search?order_by=field` could be transformed to `/search?orderBy=field&searchType=venue`.

#### Transform: Search specific parameter adapter

The guideline is, that the search specific filter adapter should be something that is easy for developers to maintain and understand.

The problem: The free text filter parameter is called “text“ in the LinkedEvents and “q“ in the Unified-Search. The UI has only 1 search field, which means that the parameter that it fills, needs to adapt to work in different searches.

The adapter should:

1. Pick the relevant URL search params; Exclude the ones that are not supported and pick the ones that could be used as they are or be transformed to a right form, so that they could be used as filters.
2. Clean the URL (as much as possible). The combined search has multiple search types and some parameters should persist even then when the search tab is not active.

The output (type `CombinedSearchAdapterOutput`) of the adapter is either `VenueSearchParams` or `EventSearchParams`.

---

An example of VenueSearchParams type, that could be used as an output for the venue search adapter:

```typescript
// NOTE: this is just an example, not an actual implementation
type VenueSearchParams = {
  q: string;
  ontologyWords: string[];
  orderBy: string;
};
```

if this would be the final active implementation of the type, the [CombinedSearchAdapterInput input type](#input-the-url-search-parameters) would transform to this like this:

1. `text` -> `q`.
2. `sportsCategories` -> `ontologyWords` (the adapter needs to also use the mapped values, not only the keys).
3. `orderBy` -> `orderBy`.

---

An example of EventSearchParams type, that could be used as an output for the event search adapter:

```typescript
// NOTE: this is just an example, not an actual implementation
type EventSearchParams = {
  text: string;
  keywords: string[];
  sort: string;
};
```

if this would be the final active implementation of the type, the [CombinedSearchAdapterInput input type](#input-the-url-search-parameters) would transform to this like this:

1. `text` -> `text`.
2. `sportsCategories` -> `keywords` (the adapter needs to also use the mapped values, not only the keys).
3. `courseOrderBy` -> `sort` & `eventOrderBy` -> `sort`.

---

#### The architecture

The URL acts as an input and a search adapter picks the params that the specific search needs and maps the query params to proper parameter field names. _In this visualization, the URL params are just some examples and may not be synced with the actual implementation._

```
┌──────────────────────────────┐     ┌────────────────────────────┐    ┌──────────────────────────────┐
| Input                        ├─────┤ Venue search adapter       ├────┤ Unified-Search               |
|==============================|     |============================|    |==============================|
| Given URL                    |     | Converted param            |    | Responds to                  |
| e.g /search?text=football&   |     | e.g /graphql?q=football    |    | param `q`                    |
| returnPath=/                 |     └────────────────────────────┘    └──────────────────────────────┘
|                              |
| Makes queries to multiple    |
| datasources                  |     ┌────────────────────────────┐    ┌──────────────────────────────┐
|                              |─────┤ Event search adapter       |────┤ LinkedEvents                 |
|                              |     |============================|    |==============================|
|                              |     | Converted param            |    | Responds to                  |
|                              |     | e.g /graphql?text=football |    | param `text`                 |
└──────────────────────────────┘     └────────────────────────────┘    └──────────────────────────────┘
```

CombinedSearchPage.tsx

```typescript
<SearchTabs initTab={initTab}>
  <CombinedSearchProvider>
    {/* The search form */}
    <SearchForm
      data-testid={searchContainerDataTestId}
      searchRoute={SEARCH_ROUTES.SEARCH}
      searchUtilities={null}
      korosBottom
      showTitle
      scrollToResultList={() => true}
    />

    {/* The search tabs, query sorters, search type switchers, etc. */}
    <SearchUtilities />

    {/* The Venue Search results */}
    <SearchTabs.Panel id="Venue">
      <VenueSearchPanel />
    </SearchTabs.Panel>

    {/* The General Event Search results */}
    <SearchTabs.Panel id={EventTypeId.General}>
      <EventSearchPanel eventType={EventTypeId.General} />
    </SearchTabs.Panel>

    {/* The Course Search results */}
    <SearchTabs.Panel id={EventTypeId.Course}>
      <EventSearchPanel eventType={EventTypeId.Course} />
    </SearchTabs.Panel>
  </CombinedSearchProvider>
</SearchTabs>
```

The `<SearchTabs/>` is there to provide a React-context / state for the tab system. The context of `<SearchTabs/>` tells

1. which tab should be the active one
2. how many search hits was found when a pre-search for the tab was done.

The `<SearchTabs/>` provider also updates the active `searchType` to the URL.

The `<CombinedSearchProvider/>` is there to have the formValues and the combined search variables stored in a React-context for each searchType. The `<CombinedSearchProvider/>` -component

1. reads the URL and uses the `CombinedSearchFormAdapter` to store the search form values in a context
2. provides the state mutator functions which can be used to update the form values in the context.
3. provides the search variables that can be used to query venues, courses and general events. The values are mapped with the `VenueSearchAdapter` and the `EventSearchAdapter` which both uses the `CombinedSearchFormAdapter` (the form values) as their input.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://tapahtumat-pr296.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by nikomakela** [comment]
The code is now ready for reaview. Please also test that the Search form works as it should. There are lots of things to remember and lot for search form to do.

- [ ] Test that all the existing search features are still working after the combined search refactoring.

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by nikomakela** [comment]
The ontologyTreeIds are currently populated wrong in the adapter. The adapter currently adds ontologyWordIds when it should add ontologyTreeIds. 

- [ ] Fix ontology tree ids queries
- [ ] Fix sportsCategories filter

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by nikomakela** [comment]
There is still an issue with the venue map search: The form is not initialized properly.

- [ ] Fix the venue map search form

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by karisal-anders** [comment]
The sports categories filtering isn't working for events and hobbies:
image

I have chosen "Tennis" as the sports category and that should filter it so that only tennis stuff is shown but the numbers don't change at all for hobbies and events.

image
Here I have chosen "Hiihto" as the sports category and the numbers for hobbies and events are the same.

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by nikomakela** [comment]
- [ ]  Add a support for internet-place filter that was in the old implementation

**Comment by nikomakela** [comment]
- [ ] Divide the URL as an input in the architecture picture. Add the form adapter in it too.

**Review by melniiv — APPROVED** [review]
Code looks good

**Comment by nikomakela** [comment]
There is actualyl stil lsome features like `getSearchFilters` that we should deprecate or even try to remove right now. They just makes everything inconsistent while they still are there, so I could try to remove then now. Lots of not so unnecessary code would be removed.

- [ ] Deprecate or remove the overlapping search form parameter handling features.

File: apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      locale
+    ] as UnifiedSearchLanguage;
+    this.q = input.text || initialVenueSearchAdapterValues.q;
+    this.ontologyTreeIds = this.getOntologyTreeIds(input);
+    this.orderByName = input.venueOrderBy
+      ? SortOrder.Ascending.toLowerCase().includes(input.venueOrderBy)
+        ? { order: SortOrder.Ascending }
+        : { order: SortOrder.Descending }
+      : initialVenueSearchAdapterValues.orderByName;
+    // TODO: this.orderByDistance = null; // input.venueOrderBy ? (SortOrder.Ascending.toLowerCase().includes(input.venueOrderBy) ? {order: SortOrder.Ascending} : {order: SortOrder.Descending}) : null;
```
Comment (karisal-anders) [inline apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts:52]: Venue search sorting is not working (And this is the code that makes it so?),
tried it in https://liikunta-pr296.dev.hel.ninja/

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

File: apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts (hunk 1 above)
Comment (nikomakela) [reply apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts]: Should be fixed now. Also the distance sorter is now implemented and the event search sorters fixed.

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Review by melniiv — APPROVED** [review]

**Review by HeikkM — APPROVED** [review]

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**